### PR TITLE
Fix issue in Dockerfile build

### PR DESCRIPTION
**Pull request by dragonsahead**
This is a subtle change in package.json and the Dockerfile to build correctly in Docker as reported by one of our customers

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -9,7 +9,12 @@ ARG MB_EDITION=oss
 WORKDIR /app/source
 
 COPY . .
-RUN NODE_ENV=production MB_EDITION=$MB_EDITION yarn --frozen-lockfile && yarn build && bin/i18n/build-translation-resources
+
+RUN yarn --frozen-lockfile
+
+ENV NODE_ENV=production
+
+RUN MB_EDITION=$MB_EDITION yarn build && bin/i18n/build-translation-resources
```
Comment (ariya) [inline Dockerfile:17]: While this works, I think a simpler alternative will be (note the newly added  `production` flag to yarn, per [its documentation](https://classic.yarnpkg.com/en/docs/cli/install/#toc-yarn-install-production-true-false)):

```
COPY . .
RUN NODE_ENV=production MB_EDITION=$MB_EDITION yarn --production=false --frozen-lockfile && yarn build && bin/i18n/build-translation-resources
```

File: package.json [hunk 2]
```diff
@@ -172,10 +172,10 @@
     "test-unit-watch": "yarn test-unit --watch",
     "test-timezones-unit": "yarn build-quick:cljs && jest --maxWorkers=2 --config jest.tz.unit.conf.json",
     "test-timezones": "yarn && ./frontend/test/__runner__/run_timezone_tests",
-    "build:js": "yarn && webpack --bail",
+    "build:js": "webpack --bail",
```
Comment (ariya) [inline package.json:175]: Is there any reason why this change is necessary?

**Comment by paoliniluis** [comment]
Closing this one in favour of https://github.com/metabase/metabase/pull/18087
